### PR TITLE
Add branch selection utility.

### PR DESCRIPTION
**Pull request by mythmon**
This is mostly a port of the equivalent functions from shield-studies-addon-utils.

### CODE REVIEW

File: src/PioneerUtils.jsm [hunk 1]
```diff
@@ -7,7 +7,8 @@ Cu.import("resource://gre/modules/XPCOMUtils.jsm");
 const { TelemetryController } = Cu.import("resource://gre/modules/TelemetryController.jsm", null);
 const { generateUUID } = Cc["@mozilla.org/uuid-generator;1"].getService(Ci.nsIUUIDGenerator);
 
-import { setCrypto, Jose, JoseJWE } from "jose-jwe-jws/dist/jose-commonjs.js";
+import { setCrypto as Jose_SetCrypto, Jose, JoseJWE } from "jose-jwe-jws/dist/jose-commonjs.js";
```
Comment (rehandalal) [inline src/PioneerUtils.jsm]: you've got `Jose_SetCrypto` here but `Jose_setCrypto` below. Maybe just got with `joseSetCrypto`?

Comment (mythmon) [reply src/PioneerUtils.jsm]: Oops, nice catch. It would probably be good to set up some linting here eventually.

File: src/PioneerUtils.jsm [hunk 2]
```diff
@@ -16,7 +17,36 @@ const PIONEER_ID_PREF = "extensions.pioneer.cachedClientID";
...
-setCrypto(crypto);
+joseSetCrypto(crypto);
+
+/**
+ * @typedef {Object} PioneerUtilsConfig
+ * @property {String} studyName
+ *   Unique name of the study.
+ *
+ * @property {String} studyVersion
+ *   Version of the study. Should match contents of install.rdf.
```
Comment (rehandalal) [inline src/PioneerUtils.jsm]: The only thing this version is used for is schema selection, it doesn't need to match install.rdf.

Comment (mythmon) [reply src/PioneerUtils.jsm]: In that case, how would you feel about making this `schemaVersion`, while internally translated it into `studyVersion` when we pass things on to Telemetry?

Comment (rehandalal) [reply src/PioneerUtils.jsm]: wfm

**Pull request merged**